### DISCUSSION TITLE
RTSP Authentication Change

### DISCUSSION THREAD

**Discussion by RUHavingFun** [mrlt8/docker-wyze-bridge#1206] [General]
I used RTSP username and password for my streams.  

            - RTSP_PATHS_ALL_READUSER=username
            - RTSP_PATHS_ALL_READPASS=password

However, now it has changed to 

            - WB_AUTH=true
            - WB_USERNAME=username
            - WB_PASSWORD=password

The problem is that I thought I could then use the username and password I just set and that is not the case.  It forces the 'wb' user and a generated extremely long key.  This is pretty hard to use as it is effectively not typeable on mobile devices where I have the streams.  I can get it to work, but it is not a very useful situation.

****  UPDATE  ****
In testing I noticed RTSP Snapshots and API Thumbnails do not require authentication.  Which would not be good.

**Comment by mrlt8** [comment]
hmm, snapshot and api thumb should be password protected. Can you try opening them in an incognito to see if they work?

All `RTSP_` prefixes have been deprecated since v2.1 over a year ago in favor of the `MTX_` prefix:
https://github.com/mrlt8/docker-wyze-bridge/commit/566b4a322a6383f846a8f1e37766333da5e3760d#diff-b35d3e7824a0338e3cda0947a6cf0178076ca65121517f05a74cca1828cad2a0

**Reply by RUHavingFun** [reply]
I just tried that, and they still work with no security.

**Reply by RUHavingFun** [reply]
Yeah, I have been using an old version.  But not that old.  I also learned today that if your RTSP viewer supplies a user name and password but the stream does not require it anymore because the parameters I thought were working were no longer working that it just ignores the credentials and the stream works.  So apparently it has been kinda broken for me for a while.

So, is there now a 'MTX_PATHS_ALL_READUSER' and 'MTX_PATHS_ALL_READPASS'?

**Reply by mrlt8** [reply]
Hmm, does it say auth is disabled in your web UI like this:
<img width="326" alt="auth" src="https://github.com/mrlt8/docker-wyze-bridge/assets/67088095/72d38381-9bf3-46de-91cd-ee32a31d8301">



You can manually change the `WB_API` by setting:
```yaml
      - WB_API=MyCustomKey
```

And yes, you can still override the stream credentials with `MTX_PATHS` prefix:
```yaml
      - MTX_PATHS_ALL_READUSER=username
      - MTX_PATHS_ALL_READPASS=password
```

**Reply by RUHavingFun** [reply]
![CleanShot 2024-05-17 at 06 41 07](https://github.com/mrlt8/docker-wyze-bridge/assets/39495965/702a05fd-2368-4b78-a393-a8502c2bd11c)

**Reply by RUHavingFun** [reply]
![CleanShot 2024-05-17 at 07 07 24](https://github.com/mrlt8/docker-wyze-bridge/assets/39495965/bcd1cf7e-78a4-4ded-94d0-86f93613a803)

**Reply by RUHavingFun** [reply]
I used these and it worked . . . kinda.  

            - WB_API=password
            - MTX_PATHS_ALL_READUSER=username
            - MTX_PATHS_ALL_READPASS=password

The links in the WebUI give me the wrong credentials, but when I change to the right ones the RTSP streams work.  The snapshot and thumbnail also seem to require credentials only when WB_AUTH=true, but there is not a lock icon, nor does it put it automatically in the link.

**Reply by mrlt8** [reply]
The auth requirement was kind of rushed for 2.9, but 2.10 should the newer auth system from media mtx.

**Reply by nikatbu** [reply]
Possibly related, is there something I'm missing with the 2.9.2 auth? After starting fresh, I'm prompted for my Wyze creds and API keys, all works seemingly and the bridge downloads my cameras, but then on the Web UI I'm presented with a user/password pop-up but nothing seems to work there (neither my Wyze user/password, nor the initial DEFAULT WB_PASSWORD. The docs talk about going to port 5000 initially, but no mention of any other cred requirements. Am I missing something obvious or is something borked?

**Reply by nikatbu** [reply]
For what it's worth, 2.8.3 doesn't have this issue... LMK if I'm spamming this thread and I can open up a new issue, etc.  Thanks.

**Reply by RUHavingFun** [reply]
When I have this set it is what I have to use to get to the WebUI.  You may want to try a private/incognito page to ensure no cached credentials problems.  If you do not set it I think it will use the default user of 'wb' and the random key it generates.

        - WB_AUTH=true
        - WB_USERNAME=username
        - WB_PASSWORD=password

**Reply by nikatbu** [reply]
Ah, cool, thanks very much! I'll read the docs a bit more closely. Thanks again!

**Reply by nikconwell** [reply]
Looks like it's defaulting to "wbadmin"

https://github.com/mrlt8/docker-wyze-bridge/blob/fb45512d9c705a6f9e8cc587cd920fafaec32a03/app/wyzebridge/config.py#L63-L67

Could also be the wyze email address but only if it's set via environment, but evidently not if it's just configured in the GUI (like I'm doing).

Thanks again for helping clear this up, much appreciated, sorry for distracting your discussion thread.  

-nik

**Comment by RUHavingFun** [comment]
This still seems to be an issue.  I updated my father's setup this weekend and found again we ran into getting the links from the WebUI were showing the "wb" user and password instead of the ones set in the YML.  Mine still do that as well.

The snapshots do require credentials when viewed in private window, but are not in the link in the UI like they are for the others.
